File: src/rb/console_util.rb
class ConsoleUtil
    def self.Prompt(question)
        puts question
        return gets
    end

    def self.DisplayMatrixResult(result)
        matrix = result.matrix
        length = [(result.side * result.side).to_s.length + 1, 2].max
        

        matrix.each { |row|
            rowText = ""
            row.each { |column|
                rowText += column.to_s.rjust(length)
            }
            puts rowText
        }
        puts "Sum: #{result.sum}"
    end
    
end

File: src/lua/consoleUtil.lua
local module = {}

function module.prompt(question)
	print(question)
	return io.read()
end

function module.displayMatrixResult(matrixResult)
	local matrix = matrixResult.matrix
	local side = #matrix
	local length = math.max(#tostring(side * side) + 1, 2)
	for row = 1, side do
		for column = 1, side do
			io.write(string.format("%" .. length .. "d", matrix[row][column]))
		end
		print()
	end
	print("Sum: " .. matrixResult.sum)
end

return module
